CombinatorialGames/Game/Concrete.lean

```diff
@@ -1,136 +1,201 @@
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Violeta Hernández Palacios
+Authors: Violeta Hernández Palacios, Junyan Xu
 -/
-import CombinatorialGames.Game.IGame
+import CombinatorialGames.Game.Loopy.IGame
 import CombinatorialGames.Game.Impartial.Basic
 
 /-!
 # Combinatorial games from a type of states
 
-A "concrete" game is a type of states endowed with well-founded move relations for the
-left and right players. This is often a more convenient representation for a game, which can then be
-used to define a `IGame`.
+In the literature, mathematicians often describe games as graphs, consisting of a set of states, as
+well as move relations for the left and right players. We define a structure `ConcreteGame` which
+facilitates this construction, bundling the left and right set functions along with the type, as
+well as functions `ConcreteGame.toLGame` and `ConcreteGame.toIGame` which turn them into the
+appropriate type of game.
+
+Mathematically, `ConcreteGame.toLGame` is nothing but the corecursor on loopy games, while
+`ConcreteGame.toIGame` is defined inductively.
+
+## Design notes
+
+When working with any "specific" game (nim, domineering, etc.) you can use  `ConcreteGame` to set up
+the basic theorems and definitions, but the intent is that you're not working with `ConcreteGame`
+directly most of the time.
+
+If you want to prove theorems about a particular game, you should organize your material as follows:
+
+- Any preliminaries about the game (definitions, structures, etc.) go in a namespace inside
+  `ConcreteGame`, e.g. `ConcreteGame.Poset`, `ConcreteGame.Domineering`, etc.
+- Theorems about the actual `IGame`s or `LGame`s should go in the corresponding namespace.
 -/
 
+universe u v
+
 noncomputable section
 
-open IGame
+open IGame Set
 
 variable {α : Type*}
 
-/-- A "concrete" game is a type of states endowed with well-founded move relations for the
-left and right players. -/
-class ConcreteGame (α : Type*) where
-  /-- The move relation for the left player. -/
-  relLeft : α → α → Prop
-  /-- The move relation for the right player. -/
-  relRight : α → α → Prop
-  /-- The move relation is well-founded. -/
-  isWellFounded_rel : IsWellFounded α fun a b ↦ relLeft a b ∨ relRight a b
+/-- A "concrete" game is a type of states endowed with move sets for the left and right players.
+
+You can use `ConcreteGame.toLGame` and `ConcreteGame.toIGame` to turn this structure into the
+appropriate game type. -/
+structure ConcreteGame (α : Type v) where
+  /-- The set of options for the left player. -/
+  leftMoves : α → Set α
+  /-- The set of options for the right player. -/
+  rightMoves : α → Set α
+  /-- To make a game in universe `u`, the set of left moves must be u-small. -/
+  small_leftMoves (a : α) : Small.{u} (leftMoves a) := by infer_instance
+  /-- To make a game in universe `u`, the set of left moves must be u-small. -/
+  small_rightMoves (a : α) : Small.{u} (rightMoves a) := by infer_instance
```

Comment:
How about we exclude these in the structure and add them as instance arguments in `toIGame/toLGame`?